src/test/kotlin/no/liflig/properties/PropertiesLoaderTest.kt

```diff
@@ -0,0 +1,83 @@
+package no.liflig.properties
+
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import java.util.Properties
+
+// By default no properties files will match the default names and be loaded
+class PropertiesLoaderTest {
+
+    @Test
+    fun `should load properties for normal runtime with SSM params`() {
+        val awsPath = "/construct/current"
+        val mockProperties = Properties().apply {
+            put("hacker.name", "Henrik")
+        }
+
+        val griidPropertiesFetcher = mockk<GriidPropertiesFetcher> {
+            every { forPrefix(awsPath) } returns mockProperties
+        }
+
+        val properties = loadPropertiesInternal(
+            applicationProperties = "testdata/application.properties",
+            griidPropertiesFetcher = griidPropertiesFetcher,
+            getenv = { awsPath }
+        )
+
+        assertEquals("Henrik", properties.getProperty("hacker.name"))
+    }
+
+    @Test
+    fun `loads invalid files`() {
+        // Java properties loading does not perform any validation
```

Comment:
```suggestion
        // Java properties loading does not perform any validation,
        // leading to some strange/unexpected results if properties being
        // loaded is not following the expected format, instead of failing.
```